servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -4767,13 +4798,23 @@ public Assertion putAssertion(ResourceContext ctx, String domainName, String pol
         }
 
         // validate to make sure we have expected values for assertion fields
+        // - and also to make sure that the associated role exists
 
         validatePolicyAssertion(assertion, caller);
 
         dbService.executePutAssertion(ctx, domainName, policyName, null, assertion, auditRef, caller);
         return assertion;
     }
 
+    private List<String> getRoleNameList(String domainName, String caller) {
+
+        List<String> names = new ArrayList<>(dbService.listRoles(domainName));
```

Comment:
there is no need to actually create yet another array list here. Having just a simple:

List<String> names = dbService.listRoles(domainName);

is sufficient since the function already creates the list and returns to us.